geoutils/georaster.py

```diff
@@ -1661,4 +1664,34 @@ def split_bands(self, copy: bool = False, subset: Optional[Union[list[int], int]
                 raster.nbands = 1
                 bands.append(raster)
 
-        return bands if len(bands) > 1 else bands[0]
\ No newline at end of file
+        return bands if len(bands) > 1 else bands[0]
+    
+    def polygonize(self, in_value=1):
+        """
+        Return a GeoDataFrame polygonized from a raster.
+        :param in_value: Value of the raster from which to create geometries (Default is 1)
+        :type in_value: int, float
+        :returns: GeoDataFrame containing the polygonized geometries
+        :rtype: geopandas.geodataframe.GeoDataFrame
+        """
+    
+        # Polygonize raster
+        if isinstance(in_value, Number):
```

Comment:
It would be nice to somehow accept a range. With floating point values, it's often hard to specify the exact value one wants, so 1 should rather be like 0.9-1.1. I that this could be done in multiple ways:

1. Accept a tuple as the in_value, where the tuple is `(min_value, max_value)`
2. Accept a list or array of exact values, where the list is `[value1, value2, value3, ...]`
3. Accept a python and/or numpy slice object: `slice(min_value, max_value)` / `np.s_[min_value, max_value]`.

Perhaps, it would be best to allow all three?

For example, I see use-cases where one wants to polygonize all non-zero/non-nan values with a field representing the pixel value. In this case, number 2 would be great to have, and probably not a too difficult thing to implement.

What do you think, @AdrienWehrle, @adehecq, @atedstone, @rhugonnet?

Reply:
It could be something like:
```python
# If it's a simple numeric value
if isinstance(in_value, Number):
    bool_msk = np.array(self.data == in_value).astype(np.uint8)
# If a tuple, assume that it's a (min_value, max_value) range.
elif isinstance(in_value, tuple):
    bool_msk = ((self.data > in_value[0]) & (self.data < in_value[1])).astype("uint8")
# If any other type of sequence (list, ndarray, etc), assume that they represent values to keep.
elif isinstance(in_value, Sequence):
    bool_msk = np.isin(self.data, in_value).astype("uint8")
```
I experimented a bit with the slicing and I guess it doesn't make sense because slicing is on indices, not on cell values. So maybe my nr 3 should be skipped!

It would also be nice to specify "all" somehow, so that all unique pixel values will be rasterized.

Reply:
Fully agree with that! And with your suggestion! 
I implemented "all" in an earlier version, but removed it at the end because raster values have to be converted to `uint8` (or equivalent) as required for the polygonization, therefore the `raster_value` column of `gdf` would only shows the integer part of the raster values... But maybe that's ok? We could link back the integers with the raster values, but I'm not sure we want such a thing...!

Reply:
What do you mean exactly? The rasterio documentation shows support for complexities up to float32 (not float64 it seems)

https://rasterio.readthedocs.io/en/latest/api/rasterio.features.html#rasterio.features.shapes

Reply:
> It would be nice to somehow accept a range. With floating point values, it's often hard to specify the exact value one wants, so 1 should rather be like 0.9-1.1. I think that this could be done in multiple ways:
> 
>     1. Accept a tuple as the in_value, where the tuple is `(min_value, max_value)`
> 
>     2. Accept a list or array of exact values, where the list is `[value1, value2, value3, ...]`
> 
>     3. Accept a python and/or numpy slice object: `slice(min_value, max_value)` / `np.s_[min_value, max_value]`.
> 
> 
> Perhaps, it would be best to allow all three?
> 
> For example, I see use-cases where one wants to polygonize all non-zero/non-nan values with a field representing the pixel value. In this case, number 2 would be great to have, and probably not a too difficult thing to implement.
> 
> What do you think, @AdrienWehrle, @adehecq, @atedstone, @rhugonnet?

A tuple with min/max as an option sounds good to me.